### ISSUE TITLE
How Role, Claim and Token is supported?

### ISSUE THREAD

**Issue by Charmatzis** [dlmelendez/identityazuretable#46]
Hi, 
first of all great job!!!! 

I just want to ask a simple question which I don't get it 100%

Here https://github.com/dlmelendez/identityazuretable/blob/master/sample/samplemvccore2/Startup.cs#L39 you are saying "if your code depends on the Role, Claim and Token … to use .AddAzureTableStores"

 If I want to use Role, Claims and Tokens should I use .AddAzureTableStores with IdentityUser?
and if my implementation is simpler is should use  .AddAzureTableStoresV2 with IdentityUserV2?
(https://github.com/dlmelendez/identityazuretable/blob/master/sample/samplemvccore2/Models/ApplicationUser.cs#L10 )

Am I getting it right? 

What's the difference between "V2" and "old version", which is obsolete?

Thanks in advance,
Christos

**Comment by dlmelendez** [comment]
@Charmatzis the older versions of the IdentityUser entity produced by the MS team has the properties of Roles, Claims and Tokens as property collections on the IdentityUser... e.g. IdentityUser.Roles, Identity.Claims, and Identity.Tokens. They made a breaking change in the IdentityUser entity removing these properties on the IdentityUser which I have paralleled with the IdentityUserV2. The IdentityUserV2 closely aligns with the latest MS entityframework IdentityUser having removed the Roles, Claims and Tokens accessors. Now, one can still take full advantage of Roles, Claims and Token features provided by the MS Identity Framework, the only difference in on IdentityUserV2 you can't persist this data on the IdentityUserV2 entity. If you are starting a new project, it is best to start with the IdentityUserV2 to keep with the same model as the MS Identity framework. I will be doing some cleanup fully deprecating the the 'old' IdentityUser and making the IdentityUserV2 the only option going forward. When MS made this change to the EntityFramework version, it was fairly abrupt and I was trying ease the transition over a few releases.
In short, if you are just starting a project use .AddAzureTableStoresV2 with IdentityUserV2. If you have an existing code base from an earlier version of the MS EntityFramework you are transitioning, use .AddAzureTableStores with IdentityUser **only** if you have heavy dependencies on the IdentityUser.Roles, IdentityUser.Claims or IdentityUser.Tokens as a first step.
Also, it should be noted, there is no data difference between the models. You can switch between 'old' and V2 at any time.
Hope this helps!
-Dave

**Comment by Charmatzis** [comment]
Wow, cool!!!!

I am starting a new apt.net core application, so I will use IdentityUserV2. So I will do the next days a PR in the samples on how to implement AddAzureTableStoresV2 with the new asp.net core template. 
How do you want to be named? e.g. samplemvccore3? under the folder /samples?

**Comment by dlmelendez** [comment]
@Charmatzis appreciate the PR! I will have a look in the next few days. There is always a ton of code with these to look at :). Closing this issue and will work on the PR when I get a few minutes.
